regparser/notice/xml.py

```diff
@@ -66,6 +66,78 @@ def _set_date_attr(self, date_type, value):
             value = value.isoformat()
         dates_tag.attrib["eregs-{}-date".format(date_type)] = value
 
+    def derive_rins(self, rins=None):
+        """
+        Modify the XML tree so that it contains meta data for regulation id
+        numbers.
+        The Federal Register API implies that documents can have more than one.
+
+        If we're not given a list, extract the information from the XML.
+
+        The XML we're adding will look something like this::
+
+            <EREGS_RINS>
+                <EREGS_RIN rin="2050-AG65" />
+            </EREGS_RINS>
+
+        :arg list rins: RINs, which should be strings.
+
+        :rtype: list
+        :returns: A list of regulation id numbers.
+        """
+        if not rins:
```

Comment:
Note that this will run if given an empty list. If that's the expected behavior, it might be worth adding that to the comment above -- "If we're not given a list or given an empty list (which indicates the meta data is faulty)..."